lib/rucio/daemons/conveyor/receiver.py

```diff
@@ -87,9 +87,7 @@ def on_message(self, frame):
            and 'issuer' in msg['job_metadata'].keys() \
            and str(msg['job_metadata']['issuer']) == str('rucio'):
 
-            if 'job_state' in msg.keys() and (
-                    str(msg['job_state']) != str('ACTIVE')
-                    or str(msg['job_state']) == str('ACTIVE') and 'job_m_replica' in msg.keys() and (str(msg['job_m_replica']).lower() == str('true'))):
+            if 'job_state' in msg.keys() and str(msg['job_state']) != str('ACTIVE'):
```

Comment:
This seems oddly written, I'd suggest:

```python
if 'job_state' in msg and msg['job_state'] != 'ACTIVE':
```

- The python docu suggst to use `key in dict` to check if a key is contained in a dictionary (http://docs.python.org/3/tutorial/datastructures.html#dictionaries)
- `'ACTIVE'` is a string already, the cast is redundant
- `msg['job_state']` should always be a string if I'm not mistaken. It makes sense to throw an error if that is not the case so we can investigate that. The cast to a string and comparison afterwards would not throw an error...
  We maybe have to test if the value is `NONE` as a special case, but I'd rather test that explicitly

Reply:
It should still work as it is. @rcarpa can check this on return, I need to merge it in the release now.